test/test_importing.py

```diff
@@ -1,31 +1,20 @@
-import pytest as pytest_collecting
-
 from .support import HPyTest
 
-# this function should probably goes somewhere into hpy.universal and/or and
-# hpy package and/or an import hook, or whatever. I do not want to think about
-# this now.
-def import_module_properly(mod):
-    raise NotImplementedError("fix me eventually")
-
-# this was moved from support.py, where it did not belong
-## class HPyLoader(ExtensionFileLoader):
-##     def create_module(self, spec):
-##         import hpy.universal
-##         return hpy.universal.load_from_spec(spec)
-
 
 class TestImporting(HPyTest):
 
-    @pytest_collecting.mark.xfail
     def test_importing_attributes(self):
+        import pytest
         import sys
-        modname = 'mytest'
-        so_filename = self.compile_module("""
+        if not hasattr(sys, "executable"):
+            pytest.skip()
+        mod = self.make_module("""
```

Comment:
`make_module` now does a proper import, so we don't need to do anything different in this test to import the module normally.

Reply:
I fear that we are stumbling across each other feet in this PR and #142, sorry for that :(.

I strongly believe that `make_module` should NOT do a proper import. Basically, the biggest difference between loading a module in memory and "properly importing" it is that the latter modifies some global state (e.g. `sys.modules`) plus it has interactions with importlib caches, etc.. And for tests, the least we mess with global state, the better it is. 
E.g. for testing the debug mode, we need to load the same module twice (with and without the debug mode support), so if we do "proper imports" we need to un-import the module between tests etc. It's a mess. That's why `self.make_module` was originally written like that, it even had a comment to make this behavior explicit ;)

See also commit ba33924: I believe that the code written this way is much cleaner and doesn't need all the hacks to sys.path, sys.modules, etc.

That said, we surely need a way to properly import modules of course, but it's probably better/easier to leave it clearly separate from the low-level stuff used by tests. So I propose the following:

1. `hpy.universal.load` is a very low-level function which just loads the module in memory, without "importing" it
2. we write a helper (maybe called `hpy.importlib` or something like that) which uses `hpy.universal.load` under the hood
3. `hpy.importlib` can have its own tests of course: these can be written by calling `self.compile_module` + `hpy.importlib.do_proper_import` instead of relying on `self.make_module` to do everything
4. the .py stubs can call `hpy.importlib` to do whatever is appropriate

What do you think about this solution?